src/clarity_vm/clarity.rs

```diff
@@ -626,6 +632,56 @@ impl<'a> ClarityBlockConnection<'a> {
         self.cost_track.unwrap()
     }
 
+    pub fn initialize_epoch_2_1(&mut self) -> Result<(), Error> {
+        let mainnet = self.mainnet;
+        let first_block_height = self.burn_state_db.get_burn_start_height();
+        let pox_prepare_length = self.burn_state_db.get_pox_prepare_length();
+        let pox_reward_cycle_length = self.burn_state_db.get_pox_reward_cycle_length();
+        let pox_rejection_fraction = self.burn_state_db.get_pox_rejection_fraction();
+
+        self.as_transaction(|tx_conn| {
+            tx_conn
+                .with_clarity_db(|db| {
+                    db.set_clarity_epoch_version(StacksEpochId::Epoch21);
+                    Ok(())
+                })
+                .unwrap();
+
+            let pox_2_code = if mainnet {
```

Comment:
I take it this will be replaced by the actual `.pox-2` code at some point?

Reply:
Yep, exactly. The next PR, which implements `stack-extend` and `delegate-stack-extend` will be the new `pox-2` code.